exp/api/v1beta1/gcpmanagedcontrolplane_types.go

```diff
@@ -27,13 +27,241 @@ const (
 	ManagedControlPlaneFinalizer = "gcpmanagedcontrolplane.infrastructure.cluster.x-k8s.io"
 )
 
+// PrivateCluster defines a private Cluster.
+type PrivateCluster struct {
+	// Enabled defines weather a cluster is private or not. If disabled cluster network is public.
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
Should we split this into nodes and endpoint?

If a user declares `privateCluster` (which is optional) in their yaml then we can assume they want a private network, so we could add a validation to the webhook to check that one of private nodes or private endpoint is true. 

We could also add a validating webhook, that is

Reply:
Good Idea!

Just to clarify, are you talking about
```go
        // ControlPlaneCidrBlock is the IP range in CIDR notation to use for the hosted master network. This range must not
	// overlap with any other ranges in use within the cluster's network. Honored when enabled is true.
	// +optional
	ControlPlaneCidrBlock string `json:"controlPlaneCidrBlock,omitempty"`

	// ControlPlaneGlobalAccess is whenever master is accessible globally or not. Honored when enabled is true.
	// +optional
	ControlPlaneGlobalAccess bool `json:"controlPlaneGlobalAccess,omitempty"`
```

these two fields in the Node and last two `ControlPlanePrivateEndpoint` & `DisableDefaultSNAT` in the Network, if I am not wrong.

Reply:
@richardcase Sorry I don't quite understand what you mean by splitting into nodes and endpoint. Can you please explain further or give an example?
The private cluster has 4 properties shown in picture out of which ContolPlaneCidrBlock is mandatory. I clubbed them into private cluster group.

![Screenshot from 2023-06-06 22-11-43](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/assets/16691314/11ebb3d8-45b1-4664-969a-3fd38159bba4)

Reply:
I'm talking about `EnablePrivateNodes` and `EnablePrivateEndpoint` on the API call: https://cloud.google.com/go/docs/reference/cloud.google.com/go/container/latest/apiv1/containerpb#cloud_google_com_go_container_apiv1_containerpb_PrivateClusterConfig